ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -671,6 +679,13 @@ private String getWorkerId() {
     return name.toString();
   }
 
+  private void updateDeltaFilesMetrics(AcidDirectory directory, String dbName, String tableName, String partName,
```

Comment:
move to the common place if possible. updateDeltaFilesMetrics could be triggered by multiple threads leading to resource starvation, think about introducing connection pool